### PR TITLE
fix: respect latest when selecting tool versions

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Adds logic to honor `latest` tag for tool version resolution.

## Context

We don't want `>=1.22.0` to pick v2 of Yarn.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/exec/buildpack.ts [hunk 1]
```diff
@@ -124,27 +141,51 @@ export async function resolveConstraint(
...
+
+  const stableVersion = releases
+    .filter((r) => isStable(r.version, versioning, pkgReleases?.tags?.latest))
+    .pop()?.version;
+  if (stableVersion) {
+    logger.warn(
+      { toolName, constraint, stableVersion },
+      'No matching tool versions found for constraint - using latest stable version'
+    );
+  }
```
Comment (viceice) [inline lib/util/exec/buildpack.ts:181]: isn't this already resolved by `stableMatchingVersion` ? Should we prefer stable before unstable?

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

File: lib/util/exec/buildpack.ts (hunk 1 above)
Comment (rarkins) [reply lib/util/exec/buildpack.ts:181]: In this case if we don't get any matching versions at all then we resort to finding the highest stable version of all releases. Can you check my added tests and see if that makes sense to you? All code is covered

File: lib/util/exec/buildpack.ts [hunk 2]
```diff
@@ -124,27 +141,51 @@ export async function resolveConstraint(
...
+
+  const highestVersion = releases.pop().version;
```
Comment (viceice) [inline lib/util/exec/buildpack.ts:183]:
```suggestion
  const highestVersion = releases.pop()!.version;
```

File: lib/util/exec/buildpack.ts (hunk 1 above)
Comment (viceice) [reply lib/util/exec/buildpack.ts:181]: I would expect this ?

```ts
 it('returns highest stable', async () => {
      expect(await resolveConstraint({ toolName: 'composer' })).toBe(
        '2.1.0'
      );
    });
```

**Review by viceice — APPROVED** [review]

**Pull request merged**